src/com/owncloud/android/lib/common/network/ChunkFromFileChannelRequestEntity.java

```diff
@@ -124,48 +129,42 @@ public void setmTransferred(long value) {
     }
 
     public void writeRequest(final OutputStream out) throws IOException {
-        int readCount = 0;
-        Iterator<OnDatatransferProgressListener> it = null;
+        long transferredBytes = 0;
+        final WritableByteChannel writeableByteChannel = Channels.newChannel(out);
 
         try {
-            mChannel.position(mOffset);
-            long size = mFile.length();
-            if (size == 0) size = -1;
             long maxCount = Math.min(mOffset + mChunkSize, mChannel.size());
-            while (mChannel.position() < maxCount) {
-                readCount = mChannel.read(mBuffer);
-                try {
-                    out.write(mBuffer.array(), 0, readCount);
-                } catch (IOException io) {
-                    // work-around try catch to filter exception in writing
-                    throw new FileRequestEntity.WriteException(io);
-                }
-                mBuffer.clear();
-                if (mTransferred < maxCount) {  // condition to avoid accumulate progress for repeated chunks
-                    mTransferred += readCount;
-                }
-                synchronized (mDataTransferListeners) {
-                    it = mDataTransferListeners.iterator();
-                    while (it.hasNext()) {
-                        it.next().onTransferProgress(readCount, mTransferred, size, mFile.getAbsolutePath());
+            Log_OC.d(TAG, "mOffset is " + mOffset + " and maxCount " + maxCount + " and contentLength: " + getContentLength());
+
+            transferredBytes = mChannel.transferTo(mOffset, getContentLength(), writeableByteChannel);
+
+            // update notification progress bar
+            if (mTransferred < maxCount) { // condition to avoid accumulate progress for repeated chunks
+                mTransferred += transferredBytes;
+            }
+            synchronized (mDataTransferListeners) {
+                for (OnDatatransferProgressListener mDataTransferListener : mDataTransferListeners) {
+                    long size = mFile.length();
+                    if (size == 0) {
+                        size = -1;
                     }
+                    mDataTransferListener.onTransferProgress(transferredBytes, mTransferred, size, mFile.getAbsolutePath());
                 }
             }
 
-        } catch (IOException io) {
+        } catch (final IOException io) {
+            Log_OC.d(TAG, "Woopsie an io ", io);
```

Comment:
As for the logging, the if/elseif/else should have the respective log statements while the last else should have some generic message :wink: